drafts/too-many-tests.md

```diff
@@ -21,19 +23,56 @@ public void CallingFunctionAbcDoesStuff()
+
+The problem is that most developers, myself included, tend to think of a class as being a unit. Any tests that you write at that level will only test part of the problem you are trying to solve. And when you come back to them 6 months down the line, the chances are you're not going to understand what they do.
+
+Imagine a requirement to manage a basket on an e-commerce website. It might look like this:
+
+```
+Given I have an empty basket
+When I add an item to the basket
+Then my basket will contain the added item
+```
+
+In order to meet this requirement you will need a module for managing baskets. This might be a microservice, or just one endpoint among many on a webservice. Regardless of how it's implemented, the observable outcome will be the same. Adding an item to the basket will cause the basket to contain that item.
+
+If we write a test at the module level then this is easy to do before writing any code. If you really want to, you can use a tool like SpecFlow which converts human readable Gherkin into test methods. Personally though, I don't think anyone ever reads those tests, so I'm quite happy to use my unit testing tool which I use for testing classes.
 
-What a waste of time!
+In writing code to make that test pass, I can make it as smelly and dirty as I want. It doesn't matter. I just need the test to pass. There might already be classes I can reuse to help me, or I can just write it all inside the public facing method which gets called.
 
-For years, companies and developers have talked about how great TDD is and sometimes about their struggles with it. The simple truth is though, that most developers have no idea what it really is!
+Once the test passes, it's refactoring time. There's no need to write any more tests because we already know it works. Any classes that are extracted should not be exposed publically. In the world of C#, this means they should be `internal`. Another thing we tend not to do!
 
-Do you remember Uncle Bob Martin's example of writing a bowling game with TDD? Instead of deciding that various classes were needed to do different things up front, he started off with a behaviour, writing a test for it and then writing code that passed it. Refactoring into classes happened later. Because these classes were internal, they didn't need their own tests. They weren't necessarily intended to be reusable. A lot of their purpose was to keep the code simple - remember the Single Responsibilty Principle?
+At this point it's ok to create dependencies inline. You might do this with a factory. They don't need to be injected into the constructor. Nothing is gained by doing this. I realise this goes against all our current practices, but the simple truth is that it's very rare for there to be multiple implementations of an interface. We've been creating all these interfaces in the name of flexibility, when actually we just don't need it.
```

Comment:
At this point, it's

Reply:
the name of flexibility when actually

Reply:
Can we have a code example here that shows what we mean to create dependencies inline?  There are a couple of ways to do this ... I might inject a factory into the constructor ... I might use a factory to create an instance that I inject into the constructor when I create the object graph ... I'm not sure the injection is the point of this part ... I think we might be talking about the fact that it is OK for our classes to take dependencies on any abstraction, not only interfaces.  Is it the proliferation of interfaces with just a single implementation that is the problem?